### PR TITLE
Adding ability to create HTML link

### PR DESCRIPTION
**Pull request by paulsbduncan**

### CODE REVIEW

File: Extension/main.css [hunk 1]
```diff
@@ -38,6 +42,12 @@ li:hover {
     background: url('email16.png') no-repeat 5px 3px transparent;
 }
 
+.html_link {
+    background-color: white;
```
Comment (paulsbduncan) [inline Extension/main.css]: When copying the HTML, we get all the cascaded style sheets with it. So, I chose the default font/font-size and bg color of Outlook.

File: Extension/background.js [hunk 2]
```diff
@@ -19,6 +19,23 @@ function snagLink(pageDetails) {
...
+  var div = document.getElementById('htmlLinkDiv');
+  div.style.display = 'block';
+  var range = document.createRange();
+  var selection = window.getSelection();
+  range.selectNodeContents(div);
+  selection.removeAllRanges();
+  selection.addRange( range );
+  document.execCommand('copy');
+  div.style.display = 'none';
+
```
Comment (blairconrad) [inline Extension/background.js]: nit: strike

File: Extension/background.js [hunk 3]
```diff
@@ -19,6 +19,23 @@ function snagLink(pageDetails) {
...
+  anchor.setAttribute("href", "http://jiraprod.agfahealthcare.com/browse/" + pageDetails.issueNumber);
+  anchor.innerHTML = pageDetails.issueNumber + ' - ' + pageDetails.title;
+
+  var div = document.getElementById('htmlLinkDiv');
+  div.style.display = 'block';
+  var range = document.createRange();
+  var selection = window.getSelection();
+  range.selectNodeContents(div);
+  selection.removeAllRanges();
+  selection.addRange( range );
```
Comment (blairconrad) [inline Extension/background.js]: While I'm nitting (see below), I find the "declare range, declare selection, do something with range, do something with selection, do something with selection and range" pattern slightly harder to read than, e.g. "declare range, do something with range, declare selection, do something with selection, do something with range and selection". The short version of that is "How do you feel about swapping lines 30 and 31? Not a huge deal.

File: Extension/main.css (hunk 1 above)
Comment (blairconrad) [reply Extension/main.css]: It's a little more complicated than that. If I paste on a blue background in Outlook, the background _stays blue_.
Your choices seem good, though. Consider adding your comment here as a comment in the code, though? There's no way I'll remember this in 11 months.

File: Extension/popup.html [hunk 4]
```diff
@@ -9,7 +9,9 @@
       <li class="action_email" id="send-email">E-mail issue</li>
       <li class="action_title" id="copy-title">Copy title</li>
       <li class="action_url" id="copy-link">Copy link as markdown</li>
+      <li class="action_htmlLink" id="copy-html-link">Copy link as HTML (for MS Office)</li>
     </ul>
+    <div id="htmlLinkDiv" class="html_link"><a id="htmlLink" href="https://jira"></a></div>
     <textarea id="ta"/>
-    </body>
```
Comment (blairconrad) [inline Extension/popup.html:14]: :+1:

**Comment by blairconrad** [comment]
Nice, @paulsbduncan! I nitted a little. I'd also squash your two commits. There's no value in seeing the html/HTML dance.

**Review by blairconrad — CHANGES_REQUESTED** [review]

**Pull request merged**